en/docs/develop/authenticating-and-authorizing-rest-apis.md

```diff
@@ -84,3 +54,47 @@ exempt_contexts=["dcr","scim2"]
     - If the user is in a secondary userstore, the incoming cert CN should be `<userstore_domain>/<username>` e.g., `SECONDARY/john`.
     - If the user is not a super tenant and belongs to the primary userstore, the incoming cert CN should be `<username@tenant_doman>` e.g., `john@abc.com`.
     - If the user is not a super tenant and belongs to a secondary userstore, the incoming cert CN should be `<userstore_domain>/<username@tenant_doman>` e.g.,             `SECONDARY/john@abc.com`.
+
+----
+
+## Scope-based authorization for REST APIs
+
+Authorization for the APIs in WSO2 Identity Server is enforced at the endpoint level using **permissions**. Each secured endpoint has a predefined minimum level of permission that is required to be able to consume the endpoint. In order to access a particular endpoint, the user has to belong to a **role** that includes the defined permissions. WSO2 Identity Server now supports scope-based API authorization for internal REST APIs.
+
+When obtaining a token to consume the API, you can define the scope corresponding to the permission required to consume the API.
+
+For example, let's assume that a user whose username is Alex, wants to retrieve the challenges available by calling the **/{user-id}/challenges** GET API available in [Challenge Question](../../develop/challenge-rest-api) REST API. This requires the user-id as an input. 
+To retrieve the challenges, Alex requires `/permission/admin/manage/identity/identitymgt/view` permission and `internal_identity_mgt_view` scope. Hence,Alex can invoke the following cURL command with `scope=internal_identity_mgt_view` and obtain a token.
+
+``` java tab="Request"
+curl -v -X POST -H "Authorization: Basic <base64encoded clientId:clientSecrect>" -k -d "grant_type=password&username=alex&password=alex123&scope=somescope" -H "Content-Type:application/x-www-form-urlencoded" https://localhost:9443/oauth2/token
+```
+
+``` java tab="Sample Request"
+curl -k -X POST -H "Authorization: Basic MUxGVzl5NERkYzZxaHVGQnBLX1JyOHA0WU1FYTpDUGl5V0hTeVp6VmJmRTFzanFNc2Vrc053Szhh" -k -d "grant_type=password&username=alex&password=alex123&scope=internal_identity_mgt_view" -H "Content-Type: application/x-www-form-urlencoded" 'https://localhost:9443/oauth2/token'
+```
+
+When the above cURL command is called, a token of the following format will be generated. If the user that requests the token has sufficient permissions to the scope defined in the request, the response will contain the scope specified in the above command. 
+
+
+``` java tab="Sample Response"
```

Comment:
Make sure the response comes as one of the tabs of the previous table. It might not work because of the paragraph before this.

Reply:
its not meant to be a tab...added the tab because it just needs to have a code block heading for consistency.